src/System.IO.FileSystem/tests/Directory/Delete.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    public class Directory_Delete_str : FileSystemTest
+    {
+        #region Utilities
+
+        public virtual void Delete(string path)
+        {
+            Directory.Delete(path);
+        }
+
+        #endregion
+
+        #region UniversalTests
+
+        [Fact]
+        public void ShouldBeAbleToDeleteHiddenDirectory()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            testDir.Attributes = FileAttributes.Hidden;
```

Comment:
Note that this is going to be a nop on Unix, where hidden is defined to mean that the name starts with a period (so setting Hidden would be a rename and thus we don't do it).  If you want to test that a hidden directory can be removed, you should probably have a test that names the directory to begin with a period, verifies (at least on Unix) that it's Hidden, and then deletes it.